File: DotNetCoreBackend/GraphQLSchema/Food/FoodQuery.cs
using GraphQL.Types;
using GraphQL.Authorization;

using DotNetCoreBackend.Services;

namespace DotNetCoreBackend.GraphQLSchema
{
    [GraphQLAuthorize(Policy = "WaiterPolicy")]
    public class FoodQuery : ObjectGraphType
    {
        public FoodQuery(IFoodService foodService)
        {
            Name = "FoodQuery";
            Description = "这里包含了有菜单";

            FieldAsync<ListGraphType<FoodType>>(
                "foods",
                arguments: new QueryArguments(
                    new QueryArgument<IntGraphType> { Name = "limit" },
                    new QueryArgument<IntGraphType> { Name = "offset" }
                    ),
                resolve: async context =>
            {
                var page = new Pageable(context, 120);

                return await foodService.GetAllFoodWithOffsetAndLimit(page.Offset, page.Limit);
            });

            FieldAsync<ListGraphType<CategoryType>>("categories", resolve: async context =>
            {
                return await foodService.GetAllCategories();
            });
        }
    }
}

File: JavaBackground/src/main/java/com/maple/graphql/query/FoodQuery.java
package com.maple.graphql.query;

import com.maple.model.Category;
import com.maple.model.Food;
import com.maple.service.FoodService;

import java.util.List;

public class FoodQuery {
    private FoodService foodService;
    
    public FoodQuery(FoodService foodService) {
        this.foodService = foodService;
    }
    
    public List<Food> foods(int limit, int offset) {
        return foodService.getFoodList(limit, offset);
    }
    
    public List<Category> categories(int limit, int offset) {
        return foodService.getCategoryList(limit, offset);
    }
}
